app/src/main/java/com/example/places/MapFragment.kt

```diff
@@ -37,6 +40,35 @@ class MapFragment : Fragment(R.layout.map_fragment), OnMapReadyCallback {
     override fun onMapReady(p0: GoogleMap) {
         p0.let {
             googleMap = it
+            viewModel.placesLiveData.observe(this, { places ->
+                markerPlaces(places)
+            })
+
+            viewModel.selectedPlace.observe(this, { place ->
+                marketPlace(place)
+            })
+
+        }
+    }
+
+    fun markerPlaces(placesList: List<Place>) {
+        for (places in placesList) {
+            if (places.latitude != null && places.longitude != null) {
+                val location = LatLng(places.latitude.toDouble(), places.longitude.toDouble())
+                googleMap.addMarker(MarkerOptions().position(location))
+                googleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(location, 10f))
+            }
+
+        }
+
+    }
+
+    fun marketPlace(place: Place) {
+        if (place.latitude != null && place.longitude != null) {
+            val location = LatLng(place.latitude.toDouble(), place.longitude.toDouble())
+            googleMap.addMarker(MarkerOptions().position(location))
```

Comment:
You should not add an extra marker for the place you selected. All the markers should already be added, you just need to zoom in to the one that was selected.